prql-compiler/src/semantic/transforms.rs

```diff
@@ -930,18 +932,34 @@ mod from_text {
             row.into_iter().map(|x| x.to_string()).collect()
         }
 
-        fn parse_row(row: csv::StringRecord) -> Vec<Literal> {
+        fn parse_row(row: csv::StringRecord) -> Result<Vec<Literal>> {
             row.into_iter()
-                .map(|x| Literal::String(x.to_string()))
-                .collect()
+                .map(|x| {
+                    // We use our own parser here, but possibly the code is no
+                    // simpler than if we wrote a mini-parser to understand
+                    // whether we need to parse a string or a number.
+                    Ok(crate::parser::parse(x)?
+                        .into_iter()
+                        .exactly_one()?
+                        .kind
+                        .as_main()
+                        .context("Expected main expr")?
+                        .as_ref()
+                        .kind
+                        .as_literal()
+                        .context("Expected literal expr")?
+                        .clone())
```

Comment:
@fusillicode I simplified this a bit. It's still not elegant, but it is a bit simpler

Reply:
@max-sixty definitely simpler and IMO even more elegant! Thanks a lot for working on it and sorry for my absence during the last couple of days 🙏